Src/StdLib/Lib/test/test_source_encoding.py

```diff
@@ -139,8 +139,12 @@ def test_error_from_string(self):
         input = "# coding: ascii\n\N{SNOWMAN}".encode('utf-8')
         with self.assertRaises(SyntaxError) as c:
             compile(input, "<string>", "exec")
-        expected = "'ascii' codec can't decode byte 0xe2 in position 16: " \
-                   "ordinal not in range(128)"
+        if sys.implementation.name == 'ironpython':
+            expected = "'us-ascii' codec can't decode byte 0xe2 in position 16: " \
```

Comment:
Is there a reason why we have `us-ascii` instead of the standard Python name `ascii`? I tracked it down to `StringOps.GetEncodingName` and it looks like it's hardcoded for the common encodings.

Reply:
It is hardcoded to `us-ascii` to make .NET Core ASCII codec have the same name as under .NET Framework 4.6. On 4.6, this code path is not reached and the name is pulled from `encoding.HeaderName` above, which produces `us-ascii`. Also codepage 28591 has `WebName` `iso-8859-1`, rather than `latin-1` as CPython would say.

I kept the .NET names preferred because these were names that IronPython was already using when I joined the project. Also, only a subset of encodings is enumerated here, the rest will simply get `cpXXXX` (on .NET Core at least).

However, I can move the switch before the .NET 4.6 part and override what .NET encoding says about itself, at least for a subset of most common encodings.